File: backend/pg_service_template/src/models/restaurant.cpp
#include "restaurant.hpp"


namespace service {


userver::formats::json::Value Serialize(
    const TRestaurant& restaurant,
    userver::formats::serialize::To<userver::formats::json::Value>
)
{
    userver::formats::json::ValueBuilder item;
    item["coordinates"] = restaurant.coordinates;
    item["name"] = restaurant.name;
    item["description"] = restaurant.description;
    item["address"] = restaurant.address;
    item["is_approved"] = restaurant.is_approved;
    item["rating"] = restaurant.rating;
    item["price_lower_bound"] = restaurant.price_lower_bound;
    item["price_upper_bound"] = restaurant.price_upper_bound;
    item["is_favorite"] = restaurant.is_favorite;

    return item.ExtractValue();
}

// TRestaurant Parse(
//     const userver::formats::json::Value& json,
//     userver::formats::parse::To<TRestaurant>
// ) = delete;


} // namespace service

File: android/core/src/main/java/model/Restaurant.kt
package model

import com.yandex.mapkit.geometry.Point

/**
 * Represents a restaurant with detailed information.
 */
data class Restaurant(
    val id: String,
    val coordinates: Point,
    val name: String,
    val description: String,
    val address: String,
    val isApproved: Boolean,
    val rating: Double,
    val priceLowerBound: Int,
    val priceUpperBound: Int,
    val openTime: String,
    val closeTime: String,
    val isFavorite: Boolean,
    val tags: List<String>,
    val inCollection: Boolean,
    val pin: String,
    val pictures: List<String>,
    val score: Long,
    val type: Pins = Pins.NONE,
    val additionalInfo: String,
)